### ISSUE TITLE
Curve encryption does not work with 0.10.0

### ISSUE THREAD

**Issue by santidhammo** [erickt/rust-zmq#357]
I am attempting to use curve encryption, it appears that libsodium is no longer used from 0.10.0, the following build commit changed this behavior:

https://github.com/erickt/rust-zmq/commit/953c9dce33bf18cea743b65f560729d70a6b3e28

However, I do not see any features which can be used to re-enable this behavior, nor can I find any documentation why this commit has been merged into main.

**Comment by magnificus** [comment]
Yeah damn a bit stuck on this, might be doing something wrong though.

let keys = CurveKeyPair::new().unwrap();

throws

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Not supported'

**Comment by magnificus** [comment]
I rolled back to 0.9.2 for now, this works for me

zmq = {version = "0.9.2", features = ["vendored"]}

**Comment by algesten** [comment]
Is there a reason #383 hasn't been merged? Can I help?

**Comment by paddor** [comment]
If you're looking for an alternative: [OMQ.rs](https://github.com/paddor/omq.rs) is a pure-Rust, ZMQ-compatible implementation with native CURVE support, including keypair generation. It keeps libzmq-style socket ergonomics without the libzmq build pain, follows the same IO-thread architecture with linear scaling, has benchmarked higher throughput and lower latency in common cases, and adds an `lz4+tcp://` compression transport.

**Comment by magnificus** [comment]
Thats cool, ill check it out
